pkg/reconciler/source/duck/duck.go

```diff
@@ -208,12 +209,17 @@ func (r *Reconciler) makeEventTypes(ctx context.Context, src *duckv1.Source) []v
 		if err != nil {
 			logging.FromContext(ctx).Warnw("Failed to parse schema as a URL", zap.String("schema", s), zap.Error(err))
 		}
+		var eventTypeNamespace string
+		if eventTypeNamespace = src.Spec.Sink.Ref.Namespace; eventTypeNamespace == "" {
+			eventTypeNamespace = defaultNamespace
+		}
```

Comment:
The defaultNamespace logic seems wrong, it should default to `src.Namespace` as opposed to the hardcoded `default` namespace